.github/workflows/test-microk8s.yaml

```diff
@@ -1,92 +1,58 @@
 name: MicroK8s Test
 
 on:
+  - push
   - pull_request
 
 jobs:
   build:
-    name: Test On AWS
+    name: Test on MicroK8s
     runs-on: ubuntu-latest
     strategy:
       matrix:
         microk8s: [stable, edge]
+        bundle: [lite, edge]
     steps:
     - name: Check out code
       uses: actions/checkout@v2
 
-    - name: Install dependencies
-      run: |
-        sudo snap install juju --classic
-        sudo snap install juju-wait --classic
-
-    - name: Bootstrap onto AWS
-      env:
-        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    - name: Install test dependencies
       run: |
         set -eux
-        /snap/bin/juju autoload-credentials --client aws
-        /snap/bin/juju bootstrap aws/us-east-1 uk8saws
-        /snap/bin/juju deploy ubuntu --constraints 'cores=4 mem=16G root-disk=60G'
-        /snap/bin/juju wait -vw
-        /snap/bin/juju status
-        /snap/bin/juju status --format yaml
+        sudo snap install charm --classic
+        sudo snap install jq
+        sudo snap install juju --classic --channel 2.7/stable
+        sudo snap install juju-helpers --classic --edge
+        sudo snap install juju-wait --classic
+        sudo snap install kubectl --classic
+        sudo snap install microk8s --classic --channel ${{ matrix.microk8s }}
+        sudo snap install yq
+        sudo apt update
+        sudo apt install -y libssl-dev python3-setuptools
+        sudo pip3 install pytest sh kfp requests pyyaml
+        sudo usermod -a -G microk8s $USER
 
-    - name: Copy code to AWS instance
+    - name: Deploy Kubeflow
       run: |
         set -eux
-        /snap/bin/juju scp -- -r $(pwd)/ ubuntu/0:~/
+        git clone git://git.launchpad.net/canonical-osm
+        cp -r canonical-osm/charms/interfaces/juju-relation-mysql mysql
+        sudo python3 ./scripts/cli.py microk8s setup
```

Comment:
This should include `--model-defaults test-mode=true`, though I find that it being slightly different from the Juju spelling (`--model-default`) a bit confusing.  Also, it would be good to be able to specify `test-mode` for the controller, which I believe has to be done with `--config test-mode=true` though it's worth testing if setting it via one or the other is sufficient.